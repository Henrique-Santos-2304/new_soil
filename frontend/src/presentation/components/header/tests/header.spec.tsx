import { screen } from '@testing-library/react'
import { Header } from '../index'
import '@testing-library/jest-dom'
import { renderThemeMock, themeContextMock } from '@/tests/index'

describe('Header Component', () => {
  it('should render component and write snapshot', () => {
    const { container } = renderThemeMock(<Header title="Olá" />)
    expect(container).toMatchSnapshot()
  })
  it('should expected header to have heading with title received', () => {
    themeContextMock('light')
    renderThemeMock(<Header title="Olá" />)

    const heading = screen.getByRole('heading', {
      level: 1,
      name: /Olá/i
    })

    expect(heading).toBeInTheDocument()
  })

  it('should expected header to have heading with subtitle received', () => {
    themeContextMock('light')
    renderThemeMock(<Header title="Olá" subTitle="send-subtitle" />)

    const heading = screen.getByRole('heading', {
      level: 4,
      name: /send-subtitle/i
    })

    expect(heading).toBeInTheDocument()
  })

  it('should expected header to have heading with content-subtitle received', () => {
    themeContextMock('light')
    renderThemeMock(<Header title="Olá" subTitle="send-content-subtitle" />)

    const heading = screen.getByRole('heading', {
      level: 4,
      name: /send-content-subtitle/i
    })

    expect(heading).toBeInTheDocument()
  })

  it('should expected header to have button logout', () => {
    themeContextMock('light')
    renderThemeMock(<Header title="Olá" subTitle="send-content-subtitle" />)

    const heading = screen.getByRole('button', { name: 'Sair' })

    expect(heading.id).toBe('btn-logout')
    expect(heading.ariaLabel).toBe('botão para fazer logout na aplicação')
    expect(heading).toBeInTheDocument()
  })
})
