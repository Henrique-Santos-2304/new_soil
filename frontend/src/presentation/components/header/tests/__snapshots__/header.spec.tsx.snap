// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render  1`] = `
<div>
  <div
    class="MuiBox-root css-imz5j4"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 MuiBox-root css-x1h9ji-MuiPaper-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1k6ks9z-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
        Sair
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root css-49rcs4"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-7inatm-MuiTypography-root"
        >
          Ol√°
        </h1>
      </div>
    </div>
  </div>
</div>
`;
